internal/scorer/scorer.go

```diff
@@ -50,32 +56,10 @@ func FromConfig(name string, r io.Reader) (*Scorer, error) {
 func (s *Scorer) Score(signals []signal.Set) float64 {
 	record := make(map[string]float64)
 	for _, s := range signals {
-		// Get all of the signal data from the set and floatify it.
+		// Get all the signal data from the set change it to a float.
 		for k, v := range signal.SetAsMap(s, true) {
-			switch r := v.(type) {
-			case float64:
-				record[k] = r
-			case float32:
-				record[k] = float64(r)
-			case int:
-				record[k] = float64(r)
-			case int16:
-				record[k] = float64(r)
-			case int32:
-				record[k] = float64(r)
-			case int64:
-				record[k] = float64(r)
-			case uint:
-				record[k] = float64(r)
-			case uint16:
-				record[k] = float64(r)
-			case uint32:
-				record[k] = float64(r)
-			case uint64:
-				record[k] = float64(r)
-			case byte:
-				record[k] = float64(r)
-			}
+			// This takes the value of v and converts it to a float64.
+			record[k] = reflect.ValueOf(v).Convert(reflect.TypeOf(float64(0))).Float()
```

Comment:
Unfortunately, this way of converting `v` to a `float64` has a subtle bug where types such as `time.Time` and `string` will cause the scoring to crash.


As it stands, I'm not sure I prefer this over the original `switch`.

The list of repeated type conversions doesn't look pretty, but it is functional and easy to understand.